.github/workflows/release.yaml

```
@@ -351,3 +362,23 @@ jobs:
               tmp/"${channel}"/* \
               '${{ env.itch_page }}':"${channel#package-}"
           done
+
+  # Deploy to Github Pages
+  deploy-to-pages:
+    needs: build
+
+    # Grant GITHUB_TOKEN the permissions required to make a Pages deployment
+    permissions:
+      pages: write # to deploy to Pages
+      id-token: write # to verify the deployment originates from an appropriate source
+
+    # Deploy to the github-pages environment
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+
+    runs-on: ubuntu-latest
```

```suggestion
    runs-on: ubuntu-latest
    needs: build
    permissions:
      # Required to upload to GitHub Pages.
      pages: write
      # Required to verify the upload originates from an appropriate source.
      id-token: write
    environment:
      name: github-pages
      url: ${{ steps.pages.outputs.page_url }}

```
I don't understand the comment on the `id-token` permission.

I'm wondering what `name: github-pages` in `environment` is needed for. I see in the example in the action's [readme](https://github.com/actions/upload-pages-artifact) that they do this too, but the action also has an _input_ called `name` that already defaults to `github-pages` and is not required.

---

Okay I was looking at the wrong action. The [`deploy-pages`](https://github.com/actions/deploy-pages) action doesn't have an input called `name`, and it does need this environment variable (although I'm not sure what it does).

---

I found an [issue](https://github.com/actions/deploy-pages/issues/329) about the `id-token` permission, it looks like a lot of people are confused about it. We should probably link to the issue in the comment.